File: amethyst.ui/src/types/events/JamLineup.ts
import { EventWithBody } from "@/hooks/EventsApiHook";
import { TeamSide } from "../TeamSide"

export class SkaterOnTrack extends EventWithBody {
    constructor(teamSide: TeamSide, position: SkaterPosition, skaterNumber: string) {
        super("SkaterOnTrack", { teamSide, skaterNumber, position });
    }
}

export class SkaterOffTrack extends EventWithBody {
    constructor(teamSide: TeamSide, skaterNumber: string) {
        super("SkaterOffTrack", { teamSide, skaterNumber });
    }
}

export class SkaterAddedToJam extends EventWithBody {
    constructor(teamSide: TeamSide, period: number, jam: number, position: SkaterPosition, skaterNumber: string) {
        super("SkaterAddedToJam", { teamSide, period, jam, position, skaterNumber });
    }
}

export class SkaterRemovedFromJam extends EventWithBody {
    constructor(teamSide: TeamSide, period: number, jam: number, skaterNumber: string) {
        super("SkaterRemovedFromJam", { teamSide, period, jam, skaterNumber });
    }
}

export enum SkaterPosition {
    Jammer = 'Jammer',
    Pivot = 'Pivot',
    Blocker = "Blocker",
}

File: amethyst/Events/JamLineup.cs
using amethyst.Domain;
using amethyst.Services;

namespace amethyst.Events;

public sealed class SkaterOnTrack(Guid7 id, SkaterOnTrackBody body) : Event<SkaterOnTrackBody>(id, body);

public sealed record SkaterOnTrackBody(TeamSide TeamSide, string? SkaterNumber, SkaterPosition Position) : TeamEventBody(TeamSide);

public enum SkaterPosition
{
    Jammer,
    Pivot,
}

